metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage.py

```diff
@@ -183,6 +205,49 @@ def _get_upstream_lineage_info(
 
         assert self._lineage_map is not None
         assert self._external_lineage_map is not None
+        if (
+            len(self._lineage_map.keys()) == 0
+            and len(self._external_lineage_map.keys()) == 0
+        ):
+            logger.debug("No lineage found.")
+            return
+
+        if self.config.include_table_lineage:
+            for dataset_name in discovered_tables:
+                if self._is_dataset_pattern_allowed(dataset_name, "table"):
+                    dataset_urn = builder.make_dataset_urn_with_platform_instance(
+                        self.platform,
+                        dataset_name,
+                        self.config.platform_instance,
+                        self.config.env,
+                    )
+                    upstream_lineage = self._get_upstream_lineage_info(dataset_name)
+                    if upstream_lineage is not None:
+                        yield self.wrap_aspect_as_workunit(
+                            "dataset", dataset_urn, "upstreamLineage", upstream_lineage
+                        )
+
+        if self.config.include_view_lineage:
+            for dataset_name in discovered_views:
```

Comment:
minor: view_name

I would recommend breaking the handling of tables and views into 2 separate sub-methods

Reply:
done